File: src/main/java/com/designpatterns/dev/behavioral/momento/java/Editor.java
package com.designpatterns.dev.behavioral.momento.java;

public class Editor {
    private  String content;

    public String getContent() {
        return this.content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public EditorState createState(){
        return new EditorState(this.content);
    }

    public void restoreState(EditorState editorState) {
        this.content = editorState.getContent();
    }
}

File: src/main/java/com/designpatterns/dev/behavioral/momento/kotlin/Editor.kt
package com.designpatterns.dev.behavioral.momento.kotlin

class Editor (var contentValue: String)  {

    fun getContent() : String{
        return contentValue
    }

    fun  createState() : EditorState {
        return  EditorState(contentValue)
    }

    fun restoreState(editorState: EditorState){
        contentValue = editorState.content;
    }
}